types/servicelimits.pp

```diff
@@ -22,10 +22,11 @@
     Optional['StartupCPUShares']    => Integer[2,262144],
     Optional['CPUQuota']            => Pattern['^([1-9][0-9]*)%$'],
     Optional['MemoryAccounting']    => Boolean,
-    Optional['MemoryLow']           => Pattern['^(\d+(K|M|G|T)?)$'],
-    Optional['MemoryHigh']          => Pattern['^(\d+(K|M|G|T)?)$'],
-    Optional['MemoryMax']           => Pattern['^(\d+(K|M|G|T)?)$'],
-    Optional['MemoryLimit']         => Pattern['^(\d+(K|M|G|T)?)$'],
+    Optional['MemoryLow']           => Pattern['^(infinity|((\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)))$'],
+    Optional['MemoryMin']           => Pattern['^(infinity|((\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)))$'],
+    Optional['MemoryHigh']          => Pattern['^(infinity|((\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)))$'],
+    Optional['MemoryMax']           => Pattern['^(infinity|((\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)))$'],
+    Optional['MemoryLimit']         => Pattern['^(infinity|((\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)))$'],
```

Comment:
I feel like some parenthesis can be removed (the original code seemed to already have redundant parenthesis):

```suggestion
    Optional['MemoryLow']           => Pattern['^(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)$'],
    Optional['MemoryMin']           => Pattern['^(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)$'],
    Optional['MemoryHigh']          => Pattern['^(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)$'],
    Optional['MemoryMax']           => Pattern['^(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)$'],
    Optional['MemoryLimit']         => Pattern['^(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)$'],
```

Also prefer `\A` to `^` and `\z` to `$` to match the beginning of the full string and not a line in the string:

```suggestion
    Optional['MemoryLow']           => Pattern['\A(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)\z'],
    Optional['MemoryMin']           => Pattern['\A(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)\z'],
    Optional['MemoryHigh']          => Pattern['\A(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)\z'],
    Optional['MemoryMax']           => Pattern['\A(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)\z'],
    Optional['MemoryLimit']         => Pattern['\A(infinity|\d+(K|M|G|T|%)(:\d+(K|M|G|T|%))?)\z'],
```

And it should be fine :+1: